projects/angular-split/src/lib/gutter/split-gutter-dynamic-injector.directive.ts

```diff
@@ -14,30 +14,31 @@ interface SplitGutterDynamicInjectorTemplateContext {
   standalone: true,
 })
 export class SplitGutterDynamicInjectorDirective {
-  @Input('asSplitGutterDynamicInjector')
-  set gutterNum(value: number) {
-    this.vcr.clear()
+  private readonly vcr = inject(ViewContainerRef)
+  private readonly templateRef = inject<TemplateRef<SplitGutterDynamicInjectorTemplateContext>>(TemplateRef)
 
-    const injector = Injector.create({
-      providers: [
-        {
-          provide: GUTTER_NUM_TOKEN,
-          useValue: value,
-        },
-      ],
-      parent: this.vcr.injector,
-    })
+  protected readonly gutterNum = input.required<number>({ alias: 'asSplitGutterDynamicInjector' })
 
-    this.vcr.createEmbeddedView(this.templateRef, { $implicit: injector })
-  }
+  constructor() {
+    effect(() => {
+      this.vcr.clear()
+
+      const injector = Injector.create({
+        providers: [
+          {
+            provide: GUTTER_NUM_TOKEN,
+            useValue: this.gutterNum(),
+          },
+        ],
+        parent: this.vcr.injector,
+      })
 
-  constructor(
-    private vcr: ViewContainerRef,
-    private templateRef: TemplateRef<SplitGutterDynamicInjectorTemplateContext>,
-  ) {}
+      this.vcr.createEmbeddedView(this.templateRef, { $implicit: injector })
+    })
```

Comment:
@SanderElias  what is your opinion on this part? Would there be a better approach to do this?

Reply:
I will be happy to. Unfortunately as it is a directive I can't use binding in the HTML so I used an effect

Reply:
Let's leave it for now I can't come up with a better alternative approach